examples/js/effects/OutlineEffect.js

```diff
@@ -26,11 +26,49 @@
 
 THREE.OutlineEffect = function ( renderer, parameters ) {
 
-	var _this = this;
+	/*
+	 * See #9918
+	 *
+	 * Here enables THREE.OutlineEffect to be called from other *Effect, like
+	 *
+	 * effect = new THREE.VREffect( new THREE.OutlineEffect( renderer ) );
+	 *
+	 * function render () {
+	 *
+ 	 * 	effect.render( scene, camera );
+	 *
+	 * }
+	 */
+	var keys = Object.keys( renderer );
+
+	for ( var i = 0, il = keys.length; i < il; i ++ ) {
+
+		var key = keys[ i ];
+
+		if ( typeof renderer[ key ] === 'function' ) {
+
+			/*
+			 * this works as
+			 * 	this.func = function ( arg1, arg2, ... ) {
+			 *
+			 * 		renderer.func( arg1, arg2, ... );
+			 *
+			 * 	};
+			 */
+			this[ key ] = renderer[ key ].bind( renderer );
+
+		} else {
+
+			// just copy property
+			this[ key ] = renderer[ key ];
+
+		}
+
+	}
```

Comment:
I added some comments, but still hard to follow?
Should I replace that with

    this.setSize = function ( width, height ) {

        renderer.setSize( width, height );

    };

    this.getSize = function () {

        return renderer.getSize();

    };

    ....

?